src/main/python/car_trecweb.py

```diff
@@ -20,22 +25,48 @@ def writer(p, fp):
     content += (u'<DOCHDR>\n')
     content += (u'\n')
     content += (u'</DOCHDR>\n')
+    if meta_dict and p.para_id in meta_dict:
+        content += (u'<TITLE>\n')
+        content += (meta_dict[p.para_id]["title"])
+        content += (u'\n</TITLE>\n')
+        content += (u'<TAG>\n')
+        content += (meta_dict[p.para_id]["tag"])
+        content += (u'\n</TAG>\n')
     content += (u'<BODY>\n')
     content += (text)
     content += (u'\n</BODY>\n')
     content += (u'</DOC>\n')
     fp.write(content)
+    
+def sanitize_string(s):
+    s = parse.unquote(s)
+    s = re.sub(r'\W+', ' ', s)
+    return s
+    
+def create_metadata_dict(tsvfile):

```

Comment:
Consider some additional validation of the file.  What if the file is there but empty? For example, a check that it has some number of desired records.

Reply:
can do, will check lines